src/test/java/io/cdap/delta/bigquery/BigQueryUtilsTest.java

```diff
@@ -153,23 +153,34 @@ public void testNormalizeTableName() {
 
     @Test
     public void testNormalizeFieldName() {
+      int maxColumnNameLength = 300;
+      String prefix = "_";
       // only contains number and letter
-      assertEquals("a2fs", BigQueryUtils.normalizeFieldName("a2fs"));
+      assertEquals("a2fs", BigQueryUtils.normalizeFieldName("a2fs", false));
       // only contains number and letter start with number
-      assertEquals("_2fas", BigQueryUtils.normalizeFieldName("2fas"));
-      // only contains number and letter and length is 128
-      String name = Strings.repeat("a1", 64);
-      assertEquals(name, BigQueryUtils.normalizeFieldName(name));
-      // only contains number and letter, starts with number and length is 128
-      name = Strings.repeat("1a", 64);
-      assertEquals("_" + name.substring(0, 127), BigQueryUtils.normalizeFieldName(name));
+      assertEquals("_2fas", BigQueryUtils.normalizeFieldName("2fas", false));
+      // only contains number and letter and length is 300
+      String name = Strings.repeat("a1", 150);
+      assertEquals(name, BigQueryUtils.normalizeFieldName(name, false));
+      // only contains number and letter, starts with number and length is 300
+      name = Strings.repeat("1a", 150);
+      assertEquals(prefix + name.substring(0, maxColumnNameLength - prefix.length()),
+              BigQueryUtils.normalizeFieldName(name, false));
       // only contains number and letter and length is 130
-      name = Strings.repeat("a1", 65);
-      assertEquals(name.substring(0, 128), BigQueryUtils.normalizeFieldName(name));
+      name = Strings.repeat("a1", 151);
+      assertEquals(name.substring(0, maxColumnNameLength), BigQueryUtils.normalizeFieldName(name, false));
       // contains invalid character
-      assertEquals("ab_c", BigQueryUtils.normalizeFieldName("ab?/c"));
+      assertEquals("ab_c", BigQueryUtils.normalizeFieldName("ab?/c", false));
       // contains space
-      assertEquals("a2_fs", BigQueryUtils.normalizeFieldName("a2 fs"));
+      assertEquals("a2_fs", BigQueryUtils.normalizeFieldName("a2 fs", false));
```

Comment:
Please add a test for whitespace comparison as well.

Reply:
Great! I've added tests for whitespace and covered all the valid and invalid characters specified in the documentation.